### DISCUSSION TITLE
Tracking Translations

### DISCUSSION THREAD

**Discussion by rg-id-me** [fnando/i18n-js#656] [Q&A] [answered]
I see we can use `I18nJS.listen` for dev environments, but for non-dev, is the idea that we should export and commit translations to source control? Or can these be auto-generated in other environments as well?

**Comment by fnando** [comment, marked as answer]
Hi! `I18nJS.listen` is meant for development. You don't have to commit translations, but you could.

The way I'm doing is generating the translations during build, then the file is imported to my JavaScript code with `import * as translations from "./config/translations"`, which is finally bundled into the final file.

**Reply by fnando** [reply]
I also wrote a migration guide (may be missing some info, so please report back if you find any issues): https://github.com/fnando/i18n-js/blob/main/MIGRATING_FROM_V3_TO_V4.md
